File: backend/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel, Field


class ProjectOut(BaseModel):
    id: str
    name: str
    status: str
    created_at: datetime
    file_count: int


class ProjectListResponse(BaseModel):
    items: list[ProjectOut]


class ProjectTreeNode(BaseModel):
    name: str
    path: str
    type: str
    file_path: str | None = None
    start_line: int | None = None
    end_line: int | None = None
    children: list["ProjectTreeNode"] = Field(default_factory=list)


class ProjectTreeResponse(BaseModel):
    project_id: str
    root: ProjectTreeNode


class SourceFileResponse(BaseModel):
    project_id: str
    path: str
    content: str


class ReviewFinding(BaseModel):
    severity: str
    title: str
    explanation: str
    file_path: str
    start_line: int | None = None
    end_line: int | None = None
    suggestion: str


class ReviewSuggestion(BaseModel):
    title: str
    detail: str


class ReviewTestCase(BaseModel):
    title: str
    detail: str


class ReviewResultResponse(BaseModel):
    project_id: str
    path: str
    target_name: str
    target_type: str
    summary: str
    findings: list[ReviewFinding]
    suggestions: list[ReviewSuggestion]
    test_cases: list[ReviewTestCase]

File: frontend/src/types/project.ts
export type Project = {
  id: string;
  name: string;
  status: string;
  created_at: string;
  file_count: number;
};

export type ProjectListResponse = {
  items: Project[];
};

export type ProjectTreeNode = {
  name: string;
  path: string;
  type: "directory" | "file" | "class" | "function";
  file_path: string | null;
  start_line: number | null;
  end_line: number | null;
  children: ProjectTreeNode[];
};

export type ProjectTreeResponse = {
  project_id: string;
  root: ProjectTreeNode;
};

export type SourceFileResponse = {
  project_id: string;
  path: string;
  content: string;
};

export type ReviewFinding = {
  severity: string;
  title: string;
  explanation: string;
  file_path: string;
  start_line: number | null;
  end_line: number | null;
  suggestion: string;
};

export type ReviewSuggestion = {
  title: string;
  detail: string;
};

export type ReviewTestCase = {
  title: string;
  detail: string;
};

export type ReviewResult = {
  project_id: string;
  path: string;
  target_name: string;
  target_type: string;
  summary: string;
  findings: ReviewFinding[];
  suggestions: ReviewSuggestion[];
  test_cases: ReviewTestCase[];
};
